code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -161,3 +161,43 @@
 	icon_state = "griffinboots"
 	item_state = "griffinboots"
 	pockets = /obj/item/weapon/storage/internal/pocket/shoes
+
+/obj/item/clothing/shoes/bhop
+	name = "jump boots"
+	desc = "A specialized pair of combat boots with a built-in propulsion system for rapid foward movement."
+	icon_state = "jetboots"
+	burn_state = FIRE_PROOF
+	pockets = /obj/item/weapon/storage/internal/pocket/shoes
+	actions_types = list(/datum/action/item_action/bhop)
+	var/recharging = FALSE
+	var/jumping = FALSE
+
+/obj/item/clothing/shoes/bhop/ui_action_click(mob/user, actiontype)
+	bhop()
+
+/obj/item/clothing/shoes/bhop/verb/bhop()
```

Comment:
Is the verb not required and can I just do that inside the ui_action_click or is there just a different way of doing so?